### PR TITLE
Feature/adv details

### PR DESCRIPTION
**Pull request by tavisca-nnarkhede**
@achirametavisca

### CODE REVIEW

File: src/app/home-module/advertisment-details/advertisment-details.component.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    this.router.params.subscribe((params: Params) => {
+      this.id = params['id'];
+      // let productMockServiceObj = new ProductMockService();
+      // productMockServiceObj.getDummyProductList().forEach((p:Product) =>{
+      //   if(p.id == params.id)
+      //   {
+      //     console.log(p);
+      //     // this.productdetails.product = p;
+      //   }
+      // })
```
Comment (achirametavisca) [inline src/app/home-module/advertisment-details/advertisment-details.component.ts]: delete this comment

File: src/app/home-module/advertisment-details/advertisment-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { ProductMockService } from "src/app/services/product-mock.service";
+
+@Component({
+  selector: 'app-advertisment-details',
+  templateUrl: './advertisment-details.component.html',
+  styleUrls: ['./advertisment-details.component.css']
+})
+export class AdvertismentDetailsComponent implements OnInit {
+  productdetails: ProductDetails;
+  id: string;
```
Comment (achirametavisca) [inline src/app/home-module/advertisment-details/advertisment-details.component.ts]: make this local variable

File: src/app/home-module/advertisment-details/advertisment-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      //     console.log(p);
+      //     // this.productdetails.product = p;
+      //   }
+      // })
+    });
+
+    this.productService.getProductDetails(this.id).subscribe(
+      (response: GetProductDetailsResponse) => {
+        this.productdetails= response.productDetails;
+        console.log(response);
```
Comment (achirametavisca) [inline src/app/home-module/advertisment-details/advertisment-details.component.ts]: remove this

**Review by achirametavisca — CHANGES_REQUESTED** [review]

File: src/app/home-module/advertisment-details/advertisment-details.component.ts (hunk 1 above)
Comment (tavisca-nnarkhede) [reply src/app/home-module/advertisment-details/advertisment-details.component.ts]: done

File: src/app/home-module/advertisment-details/advertisment-details.component.ts (hunk 2 above)
Comment (tavisca-nnarkhede) [reply src/app/home-module/advertisment-details/advertisment-details.component.ts]: done

File: src/app/home-module/advertisment-details/advertisment-details.component.ts (hunk 3 above)
Comment (tavisca-nnarkhede) [reply src/app/home-module/advertisment-details/advertisment-details.component.ts]: done

File: src/app/home-module/advertisment-details/advertisment-details.component.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component({
+  selector: 'app-advertisment-details',
+  templateUrl: './advertisment-details.component.html',
+  styleUrls: ['./advertisment-details.component.css']
+})
+export class AdvertismentDetailsComponent implements OnInit {
+  productdetails: ProductDetails;
+
+  imageArray: String[];
```
Comment (achirametavisca) [inline src/app/home-module/advertisment-details/advertisment-details.component.ts]: remove this

**Review by achirametavisca — CHANGES_REQUESTED** [review]

File: src/app/models/product-details.ts [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
 import { Product } from "./product";
-import { PagingInfo } from './paging-info';
 
 export class ProductDetails extends BaseResponse {
-    productDetails: Product;
-    pagingInfo: PagingInfo;
+    public product: Product;
+    // public title: string;
+    public location: string;
+    // public postdate: string;
```
Comment (achirametavisca) [inline src/app/models/product-details.ts]: remove comment

File: src/app/models/product-details.ts [hunk 6]
```diff
@@ -1,8 +1,11 @@
 import { BaseResponse } from "./base-response";
 import { Product } from "./product";
-import { PagingInfo } from './paging-info';
 
 export class ProductDetails extends BaseResponse {
-    productDetails: Product;
-    pagingInfo: PagingInfo;
+    public product: Product;
+    // public title: string;
```
Comment (achirametavisca) [inline src/app/models/product-details.ts]: remove comment

File: src/app/models/product-details.ts [hunk 7]
```diff
@@ -1,8 +1,11 @@
...
+    public location: string;
+    // public postdate: string;
+    public sellername: string;
```
Comment (achirametavisca) [inline src/app/models/product-details.ts]: this is coming in product itself

**Review by achirametavisca — CHANGES_REQUESTED** [review]

File: src/app/models/product-details.ts (hunk 7 above)
Comment (tavisca-nnarkhede) [reply src/app/models/product-details.ts]: done

File: src/app/models/product-details.ts (hunk 6 above)
Comment (tavisca-nnarkhede) [reply src/app/models/product-details.ts]: done

File: src/app/models/product-details.ts (hunk 5 above)
Comment (tavisca-nnarkhede) [reply src/app/models/product-details.ts]: done

**Review by achirametavisca — APPROVED** [review]

File: src/app/home-module/advertisment-details/advertisment-details.component.ts (hunk 4 above)
Comment (tavisca-nnarkhede) [reply src/app/home-module/advertisment-details/advertisment-details.component.ts]: done

**Pull request merged**